[ProtocolLib] INTERNAL ERROR: Cannot send chat message.

![image](https://user-images.githubusercontent.com/35884199/183907381-31284a8e-dc7d-4d16-abe6-c866b5d21ab8.png)

Stack Trace:
  java.lang.reflect.InvocationTargetException
        at com.comphenix.protocol.utility.ChatExtensions.sendMessageSilently(ChatExtensions.java:84)
        at com.comphenix.protocol.utility.ChatExtensions.broadcastMessageSilently(ChatExtensions.java:126)
        at com.comphenix.protocol.CommandPacket.broadcastMessageSilently(CommandPacket.java:133)
        at com.comphenix.protocol.ProtocolLib$2.publish(ProtocolLib.java:293)
        at java.logging/java.util.logging.Logger.log(Logger.java:983)
        at org.bukkit.plugin.PluginLogger.log(PluginLogger.java:34)
        at java.logging/java.util.logging.Logger.doLog(Logger.java:1010)
        at java.logging/java.util.logging.Logger.log(Logger.java:1096)
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:135)
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:99)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at com.comphenix.protocol.wrappers.AdventureComponentConverter.getComponentClass(AdventureComponentConverter.java:51)
        at com.comphenix.protocol.reflect.cloning.BukkitCloner.fromManual(BukkitCloner.java:66)
        at com.comphenix.protocol.reflect.cloning.BukkitCloner.<clinit>(BukkitCloner.java:94)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
        at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1160)
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.ensureClassInitialized(MethodHandleAccessorFactory.java:300)
        at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newConstructorAccessor(MethodHandleAccessorFactory.java:103)
        at java.base/jdk.internal.reflect.ReflectionFactory.newConstructorAccessor(ReflectionFactory.java:236)
        at java.base/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:546)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:496)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
        at com.comphenix.protocol.reflect.instances.DefaultInstances.createInstance(DefaultInstances.java:315)
        at com.comphenix.protocol.reflect.instances.DefaultInstances.getDefaultInternal(DefaultInstances.java:293)
        at com.comphenix.protocol.reflect.instances.DefaultInstances.getDefault(DefaultInstances.java:165)
        at com.comphenix.protocol.reflect.instances.DefaultInstances.create(DefaultInstances.java:333)
        at com.comphenix.protocol.reflect.cloning.AggregateCloner$Builder$1.apply(AggregateCloner.java:109)
        at com.comphenix.protocol.reflect.cloning.AggregateCloner$Builder$1.apply(AggregateCloner.java:106)
        at com.comphenix.protocol.reflect.cloning.AggregateCloner$Builder.build(AggregateCloner.java:154)
        at com.comphenix.protocol.reflect.cloning.AggregateCloner.<clinit>(AggregateCloner.java:180)
        at com.comphenix.protocol.events.PacketContainer.<clinit>(PacketContainer.java:76)
        at com.comphenix.protocol.injector.netty.manager.NetworkManagerInjector.onPacketSending(NetworkManagerInjector.java:91)
        at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector.processOutbound(NettyChannelInjector.java:525)
        at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector$2.proxyRunnable(NettyChannelInjector.java:467)
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.execute(NettyEventLoopProxy.java:220)
        at net.minecraft.network.NetworkManager.b(NetworkManager.java:211)
        at net.minecraft.network.NetworkManager.a(NetworkManager.java:191)
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1827)
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1813)
        at net.minecraft.server.level.EntityPlayer$1.a(EntityPlayer.java:261)
        at net.minecraft.world.inventory.Container.b(Container.java:196)
        at net.minecraft.world.inventory.Container.a(Container.java:176)
        at net.minecraft.server.level.EntityPlayer.a(EntityPlayer.java:557)
        at net.minecraft.server.level.EntityPlayer.g(EntityPlayer.java:561)
        at net.minecraft.server.players.PlayerList.a(PlayerList.java:388)
        at net.minecraft.server.network.LoginListener.a(LoginListener.java:207)
        at net.minecraft.server.network.LoginListener.d(LoginListener.java:193)
        at net.minecraft.server.network.LoginListener.c(LoginListener.java:85)
        at net.minecraft.network.NetworkManager.a(NetworkManager.java:271)
        at net.minecraft.server.network.ServerConnection.c(ServerConnection.java:174)
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1312)
        at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:394)
        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1197)
        at net.minecraft.server.MinecraftServer.v(MinecraftServer.java:1013)
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:293)
        at java.base/java.lang.Thread.run(Thread.java:833)
  Caused by: FieldAccessException: No field with type net.minecraft.network.chat.IChatBaseComponent exists in class ClientboundPlayerChatPacket.
        at com.comphenix.protocol.reflect.StructureModifier.writeInternal(StructureModifier.java:367)
        at com.comphenix.protocol.reflect.StructureModifier.write(StructureModifier.java:347)
        at com.comphenix.protocol.utility.ChatExtensions.createChatPackets(ChatExtensions.java:98)
        at com.comphenix.protocol.utility.ChatExtensions.sendMessageSilently(ChatExtensions.java:80)
        ... 55 more
  Dump:
  Parameters: 
    [ProtocolLib] Loaded class {0} from {1} which is not a depend or softdepend of this plugin.
    protocol.info
    manager:
      com.comphenix.protocol.injector.PacketFilterManager@3a7892a6[
        plugin=ProtocolLib v5.0.0-SNAPSHOT-b570
        server=CraftServer{serverName=CraftBukkit,serverVersion=3563-Spigot-2183145-1e3bf58,minecraftVersion=1.19.1}
        reporter=com.comphenix.protocol.ProtocolLib$1@146be410
        minecraftVersion=(MC: 1.19.1)
        asyncFilterManager=com.comphenix.protocol.async.AsyncFilterManager@3486a22b
        pluginVerifier=com.comphenix.protocol.injector.PluginVerifier@4a15ebbf
        inboundListeners=com.comphenix.protocol.injector.SortedPacketListenerList@7e97b522
        outboundListeners=com.comphenix.protocol.injector.SortedPacketListenerList@35178584
        registeredListeners=[PacketAdapter[plugin=HiddenArmor, sending=ListeningWhitelist[priority=HIGH, packets=[WINDOW_ITEMS[class=PacketPlayOutWindowItems, id=17], SET_SLOT[class=PacketPlayOutSetSlot, id=19]], gamephase=PLAYING, options=[]], receiving=EMPTY_WHITELIST], PacketAdapter[plugin=HiddenArmor, sending=ListeningWhitelist[priority=NORMAL, packets=[ENTITY_EQUIPMENT[class=PacketPlayOutEntityEquipment, id=80]], gamephase=PLAYING, options=[]], receiving=EMPTY_WHITELIST], PacketAdapter[plugin=DecentHolograms, sending=EMPTY_WHITELIST, receiving=ListeningWhitelist[priority=NORMAL, packets=[USE_ENTITY[class=PacketPlayInUseEntity, id=15]], gamephase=PLAYING, options=[]]]]
        packetInjector=com.comphenix.protocol.injector.netty.manager.NetworkManagerPacketInjector@3481eef7
        playerInjectionHandler=com.comphenix.protocol.injector.netty.manager.NetworkManagerPlayerInjector@512cb3cb
        networkManagerInjector=com.comphenix.protocol.injector.netty.manager.NetworkManagerInjector@12513269
        debug=false
        closed=false
        injected=true
      ]
  Sender:
    com.comphenix.protocol.CommandPacket@1be3be02[
      plugin=ProtocolLib v5.0.0-SNAPSHOT-b570
      logger=org.bukkit.plugin.PluginLogger@202e1ddf
      manager=com.comphenix.protocol.injector.PacketFilterManager@3a7892a6
      chatter=com.comphenix.protocol.utility.ChatExtensions@142840fa
      typeParser=com.comphenix.protocol.PacketTypeParser@ff1ea6c
      pagedMessage={}
      packetTypes=com.comphenix.protocol.concurrency.PacketTypeSet@f8dd67c
      extendedTypes=com.comphenix.protocol.concurrency.PacketTypeSet@f2fff6b
      compareTypes=com.comphenix.protocol.concurrency.PacketTypeSet@24519d06
      originalPackets={}
      listener=<null>
      compareListener=<null>
      filter=com.comphenix.protocol.CommandFilter@261b495c
      permission=protocol.admin
      name=packet
      minimumArgumentCount=1
      reporter=com.comphenix.protocol.ProtocolLib$1@146be410
    ]
  Version:
    ProtocolLib v5.0.0-SNAPSHOT-b570
  Java Version:
    18.0.1.1
  Server:
    3563-Spigot-2183145-1e3bf58 (MC: 1.19.1)

--- a/src/main/java/com/comphenix/protocol/utility/ChatExtensions.java
+++ b/src/main/java/com/comphenix/protocol/utility/ChatExtensions.java
@@ -17,15 +17,16 @@
 
 package com.comphenix.protocol.utility;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
 import com.comphenix.protocol.PacketType;
 import com.comphenix.protocol.ProtocolManager;
 import com.comphenix.protocol.events.PacketContainer;
 import com.comphenix.protocol.wrappers.EnumWrappers;
 import com.comphenix.protocol.wrappers.WrappedChatComponent;
 import com.google.common.base.Strings;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.UUID;
 import org.bukkit.Bukkit;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
@@ -56,15 +57,24 @@ public final class ChatExtensions {
 		List<PacketContainer> packets = new ArrayList<>();
 		WrappedChatComponent[] components = WrappedChatComponent.fromChatMessage(message);
 		for (WrappedChatComponent component : components) {
-			PacketContainer packet = new PacketContainer(PacketType.Play.Server.CHAT);
-			packet.getChatComponents().write(0, component);
+			PacketContainer packet;
+			if (MinecraftVersion.WILD_UPDATE.atOrAbove()) {
+				// since 1.19 system chat is extracted into a separate packet
+				packet = new PacketContainer(PacketType.Play.Server.SYSTEM_CHAT);
 
-			// 1.12+
-			packet.getChatTypes().writeSafely(0, EnumWrappers.ChatType.SYSTEM);
-			// 1.8-1.12
-			packet.getBytes().writeSafely(0, (byte) 1);
-			// 1.16+
-			packet.getUUIDs().writeSafely(0, SERVER_UUID);
+				packet.getStrings().write(0, component.getJson());
+				packet.getBooleans().write(0, false);
+			} else {
+				packet = new PacketContainer(PacketType.Play.Server.CHAT);
+				packet.getChatComponents().write(0, component);
+
+				// 1.12+
+				packet.getChatTypes().writeSafely(0, EnumWrappers.ChatType.SYSTEM);
+				// 1.8-1.12
+				packet.getBytes().writeSafely(0, (byte) 1);
+				// 1.16+
+				packet.getUUIDs().writeSafely(0, SERVER_UUID);
+			}
 
 			packets.add(packet);
 		}

